files/lib/data/conversation/ConversationAction.class.php

```diff
@@ -216,6 +216,10 @@ public function markAsRead() {
 		if (empty($this->parameters['visitTime'])) {
 			$this->parameters['visitTime'] = TIME_NOW;
 		}
+
+		if (!isset($this->parameters['userID'])) {
+			$this->parameters['userID'] = WCF::getUser()->userID;
```

Comment:
In the Infraction package I had the same issue. I forcefully `unset` the offending parameter in the `validate*` function. Usually when called through PHP the `validate*` functions are not called first and thus the parameter remains unmangled.